### DISCUSSION TITLE
What are tsg files, and how to create it?

### DISCUSSION THREAD

**Discussion by rohitcoder** [tree-sitter/tree-sitter-graph#145] [Q&A]
Hi There,

I've a small doubt, i was looking at this project and i was trying to run it, i want to get tree sitter graph from any kind of source code repo (which are supported by tree-sitter). I would like to know what is  <tsg> and how i can create a tsg file?

**Comment by hendrikvanantwerpen** [comment]
Hi @rohitcoder! Thanks for your interest in this project. I'm not sure I understand completely what you would want to use TSG for. We use it ourselves to generate stack graphs, and you can find some examples of TSG files there (e.g., https://github.com/github/stack-graphs/blob/be64a09bb3e3c229ab64855766b26dede8f4ad5c/languages/tree-sitter-stack-graphs-typescript/src/stack-graphs.tsg). Perhaps you can get some inspiration from that?

**Reply by cosmikwolf** [reply]
I think they may be referring to `tsg` referred to in the help output. The "how to use this" documentation for this project is a bit lacking. I am getting stuck trying to generate a graph of my project, and I am not sure I am doing things correctly, and I can't find any examples of how to use the CLI correctly to check what I am doing:

```
fsh ❯ tree-sitter-graph --help
tree-sitter-graph 0.12.0
Douglas Creager <dcreager@dcreager.net>
Generates graph structures from tree-sitter syntax trees

USAGE:
    tree-sitter-graph [OPTIONS] <tsg> <source>

ARGS:
    <tsg>       
    <source>    

OPTIONS:
        --allow-parse-errors    
        --global <global>       
    -h, --help                  Print help information
        --json                  
    -o, --output <output>       
    -q, --quiet                 Suppress console output
        --scope <scope>         
    -V, --version               Print version information
    -z, --lazy                  Use lazy evaluation (experimental)
```

**Reply by akshay-developer-2021** [reply]
yes, the documentation doesn't help much.

I am hard time running it. It throws the error of not finding the language grammars and instruct to init the tree-sitter.
Installed tree-sitter and cloned the grammar. Still same issue.

**Reply by Sk4467** [reply]
you need to change the 

> yes, the documentation doesn't help much.
> 
> I am hard time running it. It throws the error of not finding the language grammars and instruct to init the tree-sitter. Installed tree-sitter and cloned the grammar. Still same issue.

@akshay-developer-2021 , you need to provide path in the config file, and make sure to tune up the language-grammar as needed by the tree-sitter-graph.
